demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,390 @@
+        }
+        
+        // -------------------------------------------Filling blobs-------------------------------------------
+        auto blobDims = inputBlobsDimsInfo[inputBlobName];
+        if (blobDims.size() != 4) {
+            throw std::runtime_error("Input data does not match size of the blob");
+        }
+
+        int64 startTime = cv::getTickCount();
+        for (InferenceEngine::InferRequest& inferRequest : inferRequests) {     
+            auto inputBlob = inferRequest.GetBlob(inputBlobName);
+
+            for (unsigned i = 0; i < batchSize; i++) {
+                int imageIndex = (curPos + i) % inputImgs.size();
+                cv::Mat inputImg = inputImgs[imageIndex];
+                resizeImage(inputImg, modelInputResolution);
+
+                matU8ToBlob<uint8_t>(inputImg, inputBlob, i);
+            }
+            curPos = (curPos + batchSize) % inputImgs.size();
+
+            inferRequest.StartAsync();
+        }
+
+        // ----------------------------------------Create output info-----------------------------------------
+        int width;
+        int height;
+        std::vector<std::string> gmRowsCols = split(FLAGS_res, 'x');        
+        if (gmRowsCols.size() != 2) throw std::runtime_error("Input gmsizes key is not valid.");
+        else {
+            width = std::stoi(gmRowsCols[0]);
+            height = std::stoi(gmRowsCols[1]);
+        }
+        
+        GridMat gridMat = GridMat(inputImgs.size(), cv::Size(width, height));
+
+        cv::Mat tmpMat;
+        char key = 0;
+        unsigned fpsTestSize = 12 * batchSize;
+        double fpsSum = 0;
+        double avgFPS = 0;
+
+        for (size_t i = 0; i < fpsTestSize; i++) {
+            {
+                std::unique_lock<std::mutex> lock(mutex);
+                while (showMats.empty()) {   
+                    condVar.wait(lock);
+                }
+
+                gridMat.listUpdate(showMats);
+                fpsSum += 1. / (((cv::getTickCount() - startTime) / cv::getTickFrequency()) / framesNum);
+            }
+
+            if (!(FLAGS_no_show || FLAGS_delay == -1)) {
+                gridMat.textUpdate(fpsSum / (i + 1), true);
+                cv::imshow("main", gridMat.getMat());
+                key = static_cast<char>(cv::waitKey(1));
+            }
+        } 
+
+        avgFPS = fpsSum / fpsTestSize;
+        unsigned gridMatScale = static_cast<unsigned>(round(std::sqrt(avgFPS)));
+        gridMat = GridMat(inputImgs.size(), cv::Size(width, height), cv::Size(16, 9), gridMatScale);
+        cv::Size gridMatSize = gridMat.getSize();
+        int newDelay = ((FLAGS_delay == -1)
+                        ? avgFPS / (gridMatSize.width * (gridMatSize.height / FLAGS_b)) * 1000
+                        : FLAGS_delay);
+        unsigned fpsResultsMaxCount = 1000;
+        unsigned currentResultNum = 0;
+        double fps;
+        fpsSum = 0;
+        std::queue<double> fpsQueue;
```

Comment:
There is no need in queue

Reply:
We've already discussed it and both agreed that queue should be here in order to prevent smaller FPS values (just after start) to affect overall FPS value.